### PR TITLE
fix: improve silent message visibility 

### LINKED ISSUE #12008: send without notification is not obvious enough
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---
Attempting to send messages via "Send without notification" no longer works.
ERR... wait, is this now a toggle?

## Steps to reproduce
1. type message
2. select "Send without notification"
3. message does not send (previously it would send when you selected "Send without notification"

### Expected behaviour
The message sends, silently for the recipient.

### Actual behaviour
Now the option toggles to "Send with notification" but the user only discovers the behaviour change if selecting the `...` menu again?

image

Suggest:
* either the behaviour return to previous behaviour
* or have an indicator appear that message notification silence is now enabled.

## Talk app

**Talk app version:** (see apps admin page: `/index.php/settings/apps`)
- Nextcloud Server version 29.0.0 RC2
- Nextcloud Talk version 19.0.0-rc.2

**Custom Signaling server configured:** yes/no and version (see additional admin settings: `/index.php/index.php/settings/admin/talk#signaling_server`)

**Custom TURN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#turn_server`)

**Custom STUN server configured:** yes/no (see additional admin settings: `/index.php/settings/admin/talk#stun_server`)

## Browser

**Microphone available:** yes/no

**Camera available:** yes/no

**Operating system:** Windows/Ubuntu/...
NixOS/Linux

**Browser name:** Firefox/Chrome/...
Firefox

**Browser version:** 85/96/...
124.0.1

### Browser log

<details>
```
Insert your browser log here, this could for example include:
a) The javascript console log
b) The network log
c) ...
```

</details>

## Server configuration

**Operating system**: Ubuntu/RedHat/...

**Web server:** Apache/Nginx

**Database:** MySQL/Maria/SQLite/PostgreSQL

**PHP version:** 8.0/8.1/8.2

**Nextcloud Version:** (see admin page)
- Nextcloud Server version 29.0.0 RC2

**List of activated apps:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your server installation folder
```
</details>

**Nextcloud configuration:**

<details>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder
```
</details>

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

### PR DESCRIPTION
**Pull request by DorraJaouad**
### ☑️ Resolves

* Fix #12008

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image|
|image| image|
image| image|
image| image

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [ ] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [ ] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

### CODE REVIEW

File: src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue [hunk 1]
```diff
@@ -57,6 +57,13 @@
 						</template>
 						{{ editedDateTime }}
 					</NcActionText>
+					<!-- Silent message information -->
+					<NcActionText v-if="message.silent">
+						<template #icon>
+							<IconBellOff :size="16" />
+						</template>
+						{{ t('spreed', 'This message was sent silently') }}
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue:65]:
```suggestion
						{{ t('spreed', 'Message was sent without notification') }}
```

File: src/components/NewMessage/NewMessage.vue [hunk 2]
```diff
@@ -354,6 +357,8 @@ export default {
 				return t('spreed', 'No permission to post messages in this conversation')
 			} else if (!this.currentConversationIsJoined) {
 				return t('spreed', 'Joining conversation …')
+			} else if (this.silentChat) {
+				return t('spreed', 'Write a message (silent)')
```
Comment (DorraJaouad) [inline src/components/NewMessage/NewMessage.vue:361]:
```suggestion
				return t('spreed', 'Write a message without notification')
```

as we don't use "silent" ?

File: src/components/NewMessage/NewMessage.vue [hunk 3]
```diff
@@ -526,6 +536,15 @@ export default {
 			}
 		},
 
+		upload: {
+			immediate: true,
+			handler(value) {
+				if (value) {
+					this.silentChat = BrowserStorage.getItem('silentChat_' + this.token)
```
Comment (Antreesy) [inline src/components/NewMessage/NewMessage.vue:543]: This watcher is not needed anymore. But I would keep this line only and move to `currentUploadId` watcher:
- so if you set/reset silent toggle in UploadEditor, it got reflected in NewMessage

Comment (DorraJaouad) [reply src/components/NewMessage/NewMessage.vue:543]: Yes, that's why it was made in `upload` watcher in the first place. Isn't it better to keep in one watcher with `immediate` rather then handling the initial state in setup and the change in a watcher?

Comment (Antreesy) [reply src/components/NewMessage/NewMessage.vue:543]: `upload` is not a variable, and it's not changing with user actions. On the other hand, `currentUpload` is tracked in both inputs and actually represents opening/closing of modal

**Pull request merged**